src/App.js

```diff
@@ -21,45 +23,106 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
+  // console.log(squares);
   return squares;
 };
 
 const App = () => {
   // This starts state off as a 2D array of JS objects with
   // empty value and unique ids.
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(playerOne);
+  const [isGameOver, setGameStatus] = useState(false);
+  const [winner, setWinner] = useState(winnerInitial);
 
-  // Wave 2
-  // You will need to create a method to change the square
-  //   When it is clicked on.
-  //   Then pass it into the squares as a callback
+  const updateOnClick = (updatedSquare) => {
+    const squaresData = squares.map((row) => {
+      row.map((square) => {
+        if (square.id === updatedSquare.id) {
+          if (square.value === '') {
+            square.value = player;
+            const updatedWinner = checkForWinner();
+            if (updatedWinner) {
+              setWinner(updatedWinner);
+              setGameStatus(true);
+            }
+            setPlayer(player === playerOne ? playerTwo : playerOne);
+          }
+        }
+      });
+    });
+  };
+
+  const renderPlayerTurn = () => {
```

Comment:
Nice helper to calculate a status message. Notice that in the same way that you are calculating this message on each render, we can also do that with checkForWinner. working with the winner is a little more complex, so we could call it before the return statement and store the result in a local variable, which we could then use for calculating a message like this, or to build the winner message itself.